ql/utilities/observablevalue.hpp

```diff
@@ -66,21 +70,44 @@ namespace QuantLib {
     ObservableValue<T>::ObservableValue()
     : value_(), observable_(new Observable) {}
 
+    template <class T>
+    ObservableValue<T>::ObservableValue(T&& t)
+    : value_(std::move(t)), observable_(new Observable) {}
+
     template <class T>
     ObservableValue<T>::ObservableValue(const T& t)
     : value_(t), observable_(new Observable) {}
 
+    template <class T>
+    ObservableValue<T>::ObservableValue(ObservableValue<T>&& t) // NOLINT(performance-noexcept-move-constructor)
+    : value_(std::move(t.value_)), observable_(new Observable) {}
+
```

Comment:
I'm not sure about what we want to happen here.  If `t` is being observed by a few observers, should they start observing the newly built instance?  Until it is required, I'd probably disable this one and the corresponding assignment operator.

Reply:
The other one is making a copy and leaving the original observable alive, so any observers wouldn't know anything happened.  When moving, they would.  Then again, an rvalue is probably not likely to have any observers, so it might not be a problem.  It's tricky, though.

Reply:
Ok, I see now that `Observable` itself does not (yet) have a move constructor or a move assignment operator, and that its set of `Observer`s is not copied along with it.

As mentioned in the PR description, the first step should be to add the missing special member functions to satisfy the [Rule of Five](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all) and make the implicit behavior explicit.

For now, how about I add the move constructor and move assignment operator to both `Observable` and `ObservableValue` and just delegate to the copy constructor and copy assignment operator in each? (Deleting them is not an option because that will prevent temporaries from binding to the functions taking const lvalue reference).

In the long run, we should find a way to make `Observable` and `ObservableValue` movable because it's a bit strange in modern C++ to have copy-only types. I think it should be possible to find a way to transfer the observers to the moved-to observable. If no objections, then I can take a stab it in in the next PR.

Reply:
Ok for me.

Reply:
I saw a lot of strange errors when delegating the move constructor to the copy constructor, so for now I've decided to go with your original advice and leave them out. I've kept only the constructor and assignment operator taking the observable type by rvalue reference, and have also added a default destructor for consistency with the Rule of Five. I will try to have another look at implementing the move constructor for the class itself in a future PR.